File: SEM1/Fundamentals of programming/A8/a8-Sergiu2404-main/src/domain/movie.py


class Movie:
    def __init__(self,movie_id,title,description,genre):
        self.__movie_id=movie_id
        self.__title=title
        self.__description=description
        self.__genre=genre

    def get_id(self):
        return self.__movie_id
    def get_title(self):
        return self.__title
    def get_des(self):
        return self.__description
    def get_genre(self):
        return self.__genre

    def set_id(self,id):
        self.__movie_id=id
    def set_title(self,title):
        self.__title=title
    def set_des(self, des):
        self.__description = des
    def set_genre(self, genre):
        self.__genre =genre

    def __str__(self):
        return f"movie_id={self.__movie_id},title={self.__title},description={self.__description},genre={self.__genre}"

File: SEM2/Object Oriented Programming/A89/A9/A9/Movie.cpp
#include "Movie.h"
#include <sstream>
#include <vector>


std::vector<std::string> tokenize(std::string& string, char delimiter)
{
	std::vector<std::string> result;
	std::stringstream ss(string);
	std::string(token);

	while (getline(ss, token, delimiter))
		result.push_back(token);

	return result;
}

std::string Movie::representation()
{
	std::ostringstream output;
	output << "Title: " << this->get_title() << "; ";
	output << "Genre: " << this->get_genre() << "; ";
	output << "Year of release: " << this->get_releaseYear() << "; ";
	output << "Number of likes: " << this->get_likesNumber() << "; ";
	output << "Trailer link: " << this->get_trailer() << "\n";
	return output.str();
}

std::istream& operator>>(std::istream& inputStream, Movie movie)
{
	std::string line;
	getline(inputStream, line);

	std::vector<std::string> tokens = tokenize(line, ',');

	/*if (tokens.size() != 5)
		return inputStream;*/

	movie.set_title(tokens[0]);
	movie.set_genre(tokens[1]);
	movie.set_releaseYear(stoi(tokens[2]));
	movie.set_likesNumber(stoi(tokens[3]));
	movie.set_trailer(tokens[4]);

	return inputStream;
}

std::ostream& operator<<(std::ostream& outputStream, Movie movie)
{
	outputStream << movie.get_title() << "," << movie.get_genre() << "," << movie.get_releaseYear() << "," << movie.get_likesNumber() << "," << movie.get_trailer() << "\n";

	return outputStream;
}

